src/main/java/com/smartcodeltd/jenkinsci/plugins/buildmonitor/order/ByLastBuild.java

```diff
@@ -0,0 +1,34 @@
+package com.smartcodeltd.jenkinsci.plugins.buildmonitor.order;
+
+import hudson.model.Job;
+
+import java.util.Comparator;
+
+/**
+ * Inspired in <a href="https://github.com/Mercynary">@Mercenary</a>'s answer to
+ * issue <a href="https://github.com/jan-molak/jenkins-build-monitor-plugin/issues/113">#113</a>.
+ */
+public class ByLastBuild implements Comparator<Job<?, ?>> {
+
+    @Override
+    public int compare(Job<?, ?> a, Job<?, ?> b) {
+
+    	// none of then executed
+    	if (a.getLastBuild() == null && b.getLastBuild() == null) {
```

Comment:
Just to confirm, do you mean `getLastBuild` or `getLastCompletedBuild`? If we use `getLastBuild` the job will be moved to an earlier position in the list even if it hasn't finished running. Is that what you wanted?